src/ng/directive/ngOptions.js

```diff
@@ -348,6 +355,12 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
         var optionValuesLength = optionValuesKeys.length;
 
         for (var index = 0; index < optionValuesLength; index++) {
+          if (isString(optionValuesKeys[index]) &&
+              optionValuesKeys[index].charAt(0) === '$' &&
```

Comment:
we should cache `optionValuesKeys[index]` so that we only load the field once in unoptimized code